great_expectations/metrics/domain.py

```diff
@@ -32,6 +39,11 @@ def __new__(cls, *args, **kwargs):
             raise AbstractClassInstantiationError(cls.__name__)
         return super().__new__(cls)
 
+    @classmethod
+    @override
+    def internal_name(cls):
```

Comment:
Can we rename this to just `name`?

Reply:
yeah, i started with that. My concern is that it would be surprising that the `name` field returns something apparently unrelated, which is an implementation detail.

Reply:
This class itself is an implementation detail right?

Reply:
Oh right this will show up on a metric 🤦🏻 

I think if we need this it should be called `domain_name` then, but I would prefer to rename `Batch` to `Table` and keep the class name inference.

Reply:
we could also just move everything back into the internals of `metric`. I opted for this because i thought we would need to update more than one class string. On the other hand, I do think its generally clearer to have these Domain classes be responsible for knowing what their called, rather than the `Metric` base class.

Reply:
I am considering adding a private `_domain: DomainT` generic attribute on `Metric` for `batch.compute_metrics()` and swapping the mixin for `class TableRowCount(Metric[Table]): ...`. We can add a name override attribute if we need it later. I think the names for multi-column metrics are not consistent, so we might want to change those.

Reply:
the benefit now is that the domain params are automatically available, right? how that work with a generic, would you just need to re-implement all the domain-specific fields on the metric implementation?

Reply:
I want to turn this
```
batch_row_count = BatchRowCount(
    batch_id=...
    table=...
    min_value=...
)
batch.compute_metrics(batch_row_count)
```
into this
```
batch_row_count = BatchRowCount(min_value=...)
batch.compute_metrics(batch_row_count)
```
I think we can infer the domain arguments from the `Batch` in `batch.compute_metrics()`.

Reply:
We can't infer all domain arguments, but we can infer those first 2 I think.

Reply:
Since I'm doing childcare I can't read too closely at the moment. I prefer that we rename this from `internal_name` since, as an internal developer, it doesn't give me a sense of how this is used. I prefer `domain_name` or `domain_id` since the one usage I do see is `domain_identifier = base_type.internal_name()`.

Reply:
thanks all for the discussion. I've removed these `internal_name` methods for now in favor of the current magic string transformation in the `Metric` base class. That logic depends on our metric and domain class names following a convention that matches GX internals, which i find confusing. But if we want to change that, let's do it as a separate piece of work.